step-plans/step-plans-yaml-parser/src/main/java/step/plans/parser/yaml/schema/YamlPlanJsonSchemaGenerator.java

```diff
@@ -141,8 +140,14 @@ protected List<JsonSchemaDefinitionCreator> getDefinitionsExtensions() {
 		return extensions;
 	}
 
-	protected ArtefactFieldMetadataExtractor prepareMetadataExtractor() {
-		return new ArtefactFieldMetadataExtractor();
+	public static FieldMetadataExtractor prepareMetadataExtractor() {
+		List<FieldMetadataExtractor> metadataExtractors = new ArrayList<>();
+
+		CachedAnnotationScanner.getClassesWithAnnotation(YamlPlanReaderExtension.LOCATION, YamlPlanReaderExtension.class, Thread.currentThread().getContextClassLoader()).stream()
```

Comment:
In the future we should have only one plugin annotation that registers all the required extensions for YAML reading, Schema generation ,etc

Reply:
@jeromecomte 
I agree. Further we need to align the yaml processing for Plans and AP (make it more pluggable and based on annotations in model classes).